dttools/src/stringtools.c

```diff
@@ -46,6 +47,27 @@ char *escape_shell_string(const char *str)
 	return escaped_string;
 }
 
+char *string_escape_shell( const char *str )
+{
+	buffer_t buffer;
+	buffer_init(&buffer);
```

Comment:
I'm guessing you also want `bufer_abortonfailure` since you're not checking for (memory) errors when using the buffer.